packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -199,11 +199,15 @@ function sendAllNotifications(message, room) {
 				[notificationField]: 'mentions',
 				'u._id': { $in: mentionIdsWithoutGroups }
 			});
+		} else if ((hasMentionToAll || hasMentionToHere) && !disableAllMessageNotifications) {
+			query.$or.push({
+				[notificationField]: 'mentions'
+			});
 		}
 
 		const serverField = kind === 'email' ? 'emailNotificationMode' : `${ kind }Notifications`;
 		const serverPreference = RocketChat.settings.get(`Accounts_Default_User_Preferences_${ serverField }`);
-		if ((room.t === 'd' && serverPreference === 'mentions') || (serverPreference === 'all' && !disableAllMessageNotifications)) {
+		if ((room.t === 'd' && serverPreference === 'mentions') || ((serverPreference === 'all' || hasMentionToAll || hasMentionToHere) && !disableAllMessageNotifications)) {
```

Comment:
I think you changed the logic here I mean, 

(room.t === 'd' && serverPreference === 'mentions')  OR

Reply:
yes we do.. because `serverPreference` can be `nothing`